### ISSUE TITLE
Help required : how to add metadata in pipelines 

### ISSUE THREAD

**Issue by archanabhardwaj** [Dana-Farber-AIOS/pathml#325]
Hi 

I am working on WSI slides belongs to different sub-groups. I would like to add "labels" from my metadata sheet in order to use those information at later stage. I have few question : 

1) How can I add the "labels " from the metadata in slidedata object ? 

2) In order to run the data further in tensor-flow, I am trying to develop single dataloader representing the training object ( just like the PanNuke API training datasets).  I am mainly interested to capture WSI tiles from each "h5path" file and use further for the analysis. Here is how I extracted the array of tiles from single h5path. 

        slide1 =  h5py.File(h5_path, 'r')
       im_slide1 = slide1['tiles']['(0, 0)']['array'][...]

It gives array required for the tensorflow. But how to add WSI slide level information coming from multiple slides  ? 

Thanks in advance

**Comment by jacob-rosenthal** [comment]
`SlideData()` takes a `labels` argument which is a dict that can hold whatever slide-level labels you want. Example usage [here](https://pathml.readthedocs.io/en/latest/examples/link_loading_images.html#CODEX-spatial-proteomics). 

I'm not sure exactly what you are asking about the dataloader. I'd suggest looking at the code for `TileDataset` and using that as a reference, for example you can see there how the array for the individual tile is accessed in Line 56, tile-level labels are accessed in line 67-69, and slide-level labels are accessed in line 41-45. (Slide-level labels only need to be accessed once because they're the same for all the tiles of a slide). Then the dataloader yields all that information at once, in line 85. Hope this points you in the right direction for writing a TF dataloader

**Comment by archanabhardwaj** [comment]
Based on your suggestion, I have "labels" included for my dataset. But I am still struggling to access the "tiledata" for multiple "h5path" files and to merge into single training_set. I used TileDataset for single h5path file as given below : 
f = TileDataset("/home/Desktop/AI_pathology/multiple-preprocessed2.h5path/test1.svs.h5path")
ft = DataLoader(f,batch_size=1,   shuffle=False)
[images, masks , label, labels ]= next(iter(ft))
Is it possible to use TileDataset to read multiple h5path files present in same folder ? 
Thanks in advance

**Comment by jacob-rosenthal** [comment]
Great! Yes, I think creating a separate TileDataset for each slide and combining them with a torch ConcatDataset is probably the way to go
